tools/stub_service/src/schema_registry/service.rs

```diff
@@ -0,0 +1,318 @@
+                            .build()
+                            .expect("Get response should not fail to build");
+
+                        match get_request.complete(response).await {
+                            Ok(_) => {
+                                log::debug!(
+                                    "Get request completed successfully for Schema {}, version {}",
+                                    schema_name,
+                                    schema_version
+                                );
+                            }
+                            Err(e) => {
+                                log::error!("Failed to complete Get request: {:?}", e);
+                            }
+                        }
+                    }
+                    Err(e) => {
+                        log::error!("Error receiving Get request: {:?}", e);
+                        return Err(Box::new(e));
+                    }
+                },
+                None => {
+                    log::info!("Get command executor closed");
+                    return Ok(());
+                }
+            }
+        }
+    }
+
+    async fn put_schema_runner(
+        mut put_command_executor: service_gen::PutCommandExecutor<C>,
+        schemas: Arc<Mutex<HashMap<String, BTreeSet<Schema>>>>,
+        service_state_manager: ServiceOutputManager,
+    ) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+        loop {
+            // Wait for a new put request
+            match put_command_executor.recv().await {
+                Some(incoming_request) => match incoming_request {
+                    Ok(put_request) => {
+                        log::debug!(
+                            "Put request received: {:?}",
+                            put_request.payload.put_schema_request
+                        );
+
+                        // Extract the schema from the request
+                        let schema: Schema = put_request.payload.put_schema_request.clone().into();
+
+                        // TODO: Add verification of schema
+
+                        // Extract the schema name
+                        let schema_name = &schema.name;
+                        // Extract the schema version
+                        let schema_version: u32 = schema.version;
+
+                        // Store the schema in the HashMap
+                        {
+                            let mut schemas =
+                                schemas.lock().expect("Mutex management should be safe");
+
+                            schemas
+                                .entry(schema_name.clone())
+                                .and_modify(|schema_set| {
+                                    // Case in which the schema already exists
+
+                                    // Replace the schema with the new one or add it to the set if it doesn't exist
+                                    let old_schema = schema_set.replace(schema.clone());
```

Comment:
in this scenario, does the one that got replaced have to be exactly the same or just have the same version?